### DISCUSSION TITLE
MacOS v12.2.1 Monterey, problems installing host software

### DISCUSSION THREAD

**Discussion by rogerbratseth** [keirf/greaseweazle#185] [Q&A]
I get this terrible error message when I try to install on my Mac, anyone else seen this? Have tested both methods described here in the repo, and get the same error:

% python3 -m pip install greaseweazle-tools-0.40/greaseweazle-0.40.tar.gz
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Processing ./greaseweazle-tools-0.40/greaseweazle-0.40.tar.gz
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/qj/9bw17n6n1pd9_0p3n21wvsl40000gn/T/pip-req-build-ntomm7y8/setup.py", line 2, in <module>
          from setuptools_scm import get_version
      ModuleNotFoundError: No module named 'setuptools_scm'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
%

**Comment by keirf** [comment]
Try `python3 -m pip install setuptools_scm`. Alternatively investigate setting up pyenv and virtualenv in macos. This gets you a private and modern-python-friendly environment.

**Reply by rogerbratseth** [reply]
It seems I got a bit further along, but still some errors. Hope it´s okay that I copy my output to the discussion:

rogerbratseth@Rogers-iMac Downloads % python3 -m pip install setuptools_scm
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Collecting setuptools_scm
  Downloading setuptools_scm-6.4.2-py3-none-any.whl (37 kB)
Collecting tomli>=1.0.0
  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting packaging>=20.0
  Downloading packaging-21.3-py3-none-any.whl (40 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.8/40.8 KB 2.3 MB/s eta 0:00:00
Requirement already satisfied: setuptools in /usr/local/lib/python3.9/site-packages (from setuptools_scm) (59.0.1)
Collecting pyparsing!=3.0.5,>=2.0.2
  Downloading pyparsing-3.0.7-py3-none-any.whl (98 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.0/98.0 KB 4.9 MB/s eta 0:00:00
Installing collected packages: tomli, pyparsing, packaging, setuptools_scm
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Successfully installed packaging-21.3 pyparsing-3.0.7 setuptools_scm-6.4.2 tomli-2.0.1


rogerbratseth@Rogers-iMac Downloads % python3 -m pip install greaseweazle-tools-0.40/greaseweazle-0.40.tar.gz


DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Processing ./greaseweazle-tools-0.40/greaseweazle-0.40.tar.gz
  Preparing metadata (setup.py) ... done
Collecting crcmod
  Downloading crcmod-1.7.tar.gz (89 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.7/89.7 KB 4.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting bitarray
  Downloading bitarray-2.4.1-cp39-cp39-macosx_10_9_x86_64.whl (98 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.4/98.4 KB 4.2 MB/s eta 0:00:00
Collecting pyserial
  Downloading pyserial-3.5-py2.py3-none-any.whl (90 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.6/90.6 KB 6.2 MB/s eta 0:00:00
Collecting requests
  Downloading requests-2.27.1-py2.py3-none-any.whl (63 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.1/63.1 KB 3.5 MB/s eta 0:00:00
Collecting certifi>=2017.4.17
  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 149.2/149.2 KB 9.4 MB/s eta 0:00:00
Collecting idna<4,>=2.5
  Downloading idna-3.3-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.2/61.2 KB 3.7 MB/s eta 0:00:00
Collecting charset-normalizer~=2.0.0
  Downloading charset_normalizer-2.0.12-py3-none-any.whl (39 kB)
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.9-py2.py3-none-any.whl (138 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 139.0/139.0 KB 6.8 MB/s eta 0:00:00
Building wheels for collected packages: greaseweazle, crcmod
  Building wheel for greaseweazle (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [62 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-11-x86_64-3.9
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/error.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/track.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/cli.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/usb.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/flux.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/update.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/pin.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/clean.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/write.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/delays.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/util.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/convert.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/reset.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/bandwidth.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/rpm.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/erase.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/info.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/seek.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/read.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/codec
      copying src/greaseweazle/codec/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec
      copying src/greaseweazle/codec/formats.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/sf7.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/ipf.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/edsk.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/kryoflux.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/img.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/d81.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/hfe.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/acorn.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/image.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/scp.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/adf.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/optimised
      copying src/greaseweazle/optimised/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/optimised
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/ibm
      copying src/greaseweazle/codec/ibm/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/ibm
      copying src/greaseweazle/codec/ibm/fm.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/ibm
      copying src/greaseweazle/codec/ibm/mfm.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/ibm
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/amiga
      copying src/greaseweazle/codec/amiga/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/amiga
      copying src/greaseweazle/codec/amiga/amigados.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/amiga
      running build_ext
      building 'greaseweazle.optimised.optimised' extension
      creating build/temp.macosx-11-x86_64-3.9
      creating build/temp.macosx-11-x86_64-3.9/src
      creating build/temp.macosx-11-x86_64-3.9/src/greaseweazle
      creating build/temp.macosx-11-x86_64-3.9/src/greaseweazle/optimised
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c src/greaseweazle/optimised/optimised.c -o build/temp.macosx-11-x86_64-3.9/src/greaseweazle/optimised/optimised.o
      xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for greaseweazle
  Running setup.py clean for greaseweazle
  Building wheel for crcmod (setup.py) ... done
  Created wheel for crcmod: filename=crcmod-1.7-py3-none-any.whl size=18850 sha256=f4205415033c3dd6cfe4a971bbd03142a17543a4c75158c532c64105a1a46c6d
  Stored in directory: /Users/rogerbratseth/Library/Caches/pip/wheels/4a/6c/a6/ffdd136310039bf226f2707a9a8e6857be7d70a3fc061f6b36
Successfully built crcmod
Failed to build greaseweazle
Installing collected packages: pyserial, crcmod, certifi, bitarray, urllib3, idna, charset-normalizer, requests, greaseweazle
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
  Running setup.py install for greaseweazle ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for greaseweazle did not run successfully.
  │ exit code: 1
  ╰─> [64 lines of output]
      running install
      /usr/local/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.macosx-11-x86_64-3.9
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/error.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/track.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/cli.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/usb.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      copying src/greaseweazle/flux.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/update.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/pin.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/clean.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/write.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/delays.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/util.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/convert.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/reset.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/bandwidth.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/rpm.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/erase.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/info.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/seek.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      copying src/greaseweazle/tools/read.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/tools
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/codec
      copying src/greaseweazle/codec/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec
      copying src/greaseweazle/codec/formats.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/sf7.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/ipf.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/edsk.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/kryoflux.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/img.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/d81.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/hfe.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/acorn.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/image.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/scp.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      copying src/greaseweazle/image/adf.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/image
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/optimised
      copying src/greaseweazle/optimised/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/optimised
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/ibm
      copying src/greaseweazle/codec/ibm/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/ibm
      copying src/greaseweazle/codec/ibm/fm.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/ibm
      copying src/greaseweazle/codec/ibm/mfm.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/ibm
      creating build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/amiga
      copying src/greaseweazle/codec/amiga/__init__.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/amiga
      copying src/greaseweazle/codec/amiga/amigados.py -> build/lib.macosx-11-x86_64-3.9/greaseweazle/codec/amiga
      running build_ext
      building 'greaseweazle.optimised.optimised' extension
      creating build/temp.macosx-11-x86_64-3.9
      creating build/temp.macosx-11-x86_64-3.9/src
      creating build/temp.macosx-11-x86_64-3.9/src/greaseweazle
      creating build/temp.macosx-11-x86_64-3.9/src/greaseweazle/optimised
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c src/greaseweazle/optimised/optimised.c -o build/temp.macosx-11-x86_64-3.9/src/greaseweazle/optimised/optimised.o
      xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> greaseweazle

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
rogerbratseth@Rogers-iMac Downloads %

**Comment by keirf** [comment]
Or it looks like the answer might be to install Python 3.10 through Homebrew. Could you try that?

**Reply by rogerbratseth** [reply]
sure!

**Reply by keirf** [reply]
I think this is a combination of Homebrew Python 3.9 and the fact I don't have a totally modern build config for Python and also you need to install X Code as the C compiler is currently missing in your setup.

**Reply by rogerbratseth** [reply]
Was missing the "Apple Command Line", took a while to install...

**Reply by rogerbratseth** [reply]
Success now I think:

rogerbratseth@Rogers-iMac Downloads % python3 -m pip install greaseweazle-tools-0.40/greaseweazle-0.40.tar.gz
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Processing ./greaseweazle-tools-0.40/greaseweazle-0.40.tar.gz
  Preparing metadata (setup.py) ... done
Requirement already satisfied: crcmod in /usr/local/lib/python3.9/site-packages (from greaseweazle==0.40) (1.7)
Requirement already satisfied: bitarray in /usr/local/lib/python3.9/site-packages (from greaseweazle==0.40) (2.4.1)
Requirement already satisfied: pyserial in /usr/local/lib/python3.9/site-packages (from greaseweazle==0.40) (3.5)
Requirement already satisfied: requests in /usr/local/lib/python3.9/site-packages (from greaseweazle==0.40) (2.27.1)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.9/site-packages (from requests->greaseweazle==0.40) (2021.10.8)
Requirement already satisfied: charset-normalizer~=2.0.0 in /usr/local/lib/python3.9/site-packages (from requests->greaseweazle==0.40) (2.0.12)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.9/site-packages (from requests->greaseweazle==0.40) (1.26.9)
Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.9/site-packages (from requests->greaseweazle==0.40) (3.3)
Building wheels for collected packages: greaseweazle
  Building wheel for greaseweazle (setup.py) ... done
  Created wheel for greaseweazle: filename=greaseweazle-0.40-cp39-cp39-macosx_12_0_x86_64.whl size=79896 sha256=48efa58616b5334ce5c399f8770dd9cb2f890dfd71fc80ca2dd070b1d6776bb2
  Stored in directory: /Users/rogerbratseth/Library/Caches/pip/wheels/25/9a/d9/def30a41abd000b3a3ef81023f82767300a3e114ffb8ee67f8
Successfully built greaseweazle
Installing collected packages: greaseweazle
  DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Successfully installed greaseweazle-0.40
rogerbratseth@Rogers-iMac Downloads %

**Reply by rogerbratseth** [reply]
Thank you very much, this was a good session! Now I need to wait for the mailman to deliver the hardware :-)

**Reply by rogerbratseth** [reply]
So to summarize, I had to do the following:
% xcode-select --install
% brew install python@3.10
% python3 -m pip install setuptools_scm
% python3 -m pip install Downloads/greaseweazle-tools-0.40/greaseweazle-0.40.tar.gz

**Reply by keirf** [reply]
Bingo, I think `xcode-select --install` is key here. Installing Python 3.10 also looks sensible. I also needed to update my package metadata. So a few fixes to code and documentation!

**Reply by keirf** [reply]
All fixed now, shiny new v0.41 release and updated wiki instructions. Thanks for the report!
